grails-app/services/org/pih/warehouse/data/ProductSynonymDataService.groovy

```diff
@@ -56,16 +59,18 @@ class ProductSynonymDataService {
                 } else {
                     command.errors.reject("Row ${index + 1}: Locale '${params['locale']}' is not a supported locale")
                 }
+
             }
-        }
-    }
 
-    void importData(ImportDataCommand command) {
-        log.info "Import data " + command.filename
+            // If none validation error occured yet, try to add a synonym
+            if (!command.errors.allErrors) {
+                Synonym synonym = new Synonym(name: params['name'], locale: new Locale(params['locale']), synonymTypeCode: params['synonymTypeCode'])
+                product.addToSynonyms(synonym)
+                if (!synonym.validate() || !product.save(flush: true, failOnError: true)) {
+                    command.errors.reject("Row ${index + 1}: Validation error occured. Most probably you are trying to add multiple DISPLAY_NAME synonym for one locale")
```

Comment:
I don't like the "most probably" type of messages. Normally, I think you should have access to `synonym.errors`, however, in this case, please bring back the `validateData` and do rejects based on the real cause, not the "most probable" one.